core/src/main/java/com/linecorp/armeria/server/auth/HttpAuthService.java

```diff
@@ -20,79 +20,51 @@
 
 import java.util.function.Function;
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
 import com.google.common.collect.ImmutableList;
 
 import com.linecorp.armeria.common.HttpRequest;
 import com.linecorp.armeria.common.HttpResponse;
-import com.linecorp.armeria.common.util.Exceptions;
 import com.linecorp.armeria.server.HttpService;
-import com.linecorp.armeria.server.ServiceRequestContext;
-import com.linecorp.armeria.server.SimpleDecoratingHttpService;
 
 /**
  * Decorates an {@link HttpService} to provide HTTP authorization functionality.
  *
- * @see HttpAuthServiceBuilder
+ * @deprecated Use {@link AuthService}.
  */
-public final class HttpAuthService extends SimpleDecoratingHttpService {
-
-    static final Logger logger = LoggerFactory.getLogger(HttpAuthService.class);
+@Deprecated
+public final class HttpAuthService extends AuthService {
 
     /**
      * Creates a new HTTP authorization {@link HttpService} decorator using the specified
      * {@link Authorizer}s.
      *
      * @param authorizers a list of {@link Authorizer}s.
+     * @deprecated Use {@link AuthService#newDecorator(Iterable)}.
      */
-    public static Function<? super HttpService, HttpAuthService> newDecorator(
+    @Deprecated
+    public static Function<? super HttpService, AuthService> newDecorator(
```

Comment:
Can we change the return type without breaking change?

Reply:
I tried but in order to do that I need to copy the whole source or add another type parameter to the `AuthService`. So I have chosen this way because most of the users:
- do not declare a variable for the return value
- if they do that, they can easily notice because the IDE shows which one is wrong
So let me just do this way. 🙏

Reply:
I understood. Then how about adding this to the breaking change list?

Reply:
Sure 😉